plugins/soundsourcem4a/audiosourcem4a.cpp

```diff
@@ -0,0 +1,307 @@
+#include "audiosourcem4a.h"
+
+#include "sampleutil.h"
+
+#ifdef __WINDOWS__
+#include <io.h>
+#include <fcntl.h>
+#endif
+
+#ifdef _MSC_VER
+#define S_ISDIR(mode) (mode & _S_IFDIR)
+#define strcasecmp stricmp
+#define strncasecmp strnicmp
+#endif
+
+#ifdef __M4AHACK__
+typedef uint32_t SAMPLERATE_TYPE;
+#else
+typedef unsigned long SAMPLERATE_TYPE;
+#endif
+
+namespace Mixxx {
+
+namespace
+{
+
+// AAC: "... each block decodes to 1024 time-domain samples."
+const AudioSource::size_type kFramesPerSampleBlock = 1024;
+
+// MP4SampleId is 1-based
+const MP4SampleId kMinSampleId = 1;
+
+// Decoding will be restarted one or more blocks of samples
+// before the actual position to avoid audible glitches.
+// Two blocks of samples seems to be enough here.
+const MP4SampleId kSampleIdPrefetchCount = 2;
+
+MP4TrackId findAacTrackId(MP4FileHandle hFile) {
```

Comment:
This is just a question about coding style. We have no strict rule that is against a free function but I think we have nearly non of them elsewhere in Mixxx. It jumps to my eyes, just because we normally have the constructor at the top of a cpp file. This is also no strong argument though. Just to save you from extra work, leave it there.  

By the way, thank you for all the work. As far as I am with reading, it is already really a big improvement in design a readability. And sorry in advance, if I am sometime too niggling compared to the original code state where we come from.